src/Symfony/Component/HttpKernel/Controller/ArgumentResolver/RequestPayloadValueResolver.php

```diff
@@ -185,7 +185,7 @@ public static function getSubscribedEvents(): array
 
     private function mapQueryString(Request $request, ArgumentMetadata $argument, MapQueryString $attribute): ?object
     {
-        if (!$data = $request->query->all()) {
+        if (!($data = $request->query->all()) && ($attribute->metadata->isNullable() || $attribute->metadata->hasDefaultValue())) {
```

Comment:
Note that in 7.2 the metadata argument is passed to this method, so this statement can be shortened:
```suggestion
        if (!($data = $request->query->all()) && ($argument->isNullable() || $argument->hasDefaultValue())) {
```